src/main.rs

```diff
@@ -0,0 +1,59 @@
+use std::{
+    fs::{read_to_string, File},
+    io::{Read, Write},
+    path::PathBuf,
+};
+
+use clap::Parser;
+use luci::{execution::Executable, messages::Messages, scenario::Scenario};
+
+#[derive(Parser, Debug)]
+#[command(
+    name = "luci",
+    about = "Generate a Graphviz DOT graph from a scenario description."
+)]
+struct Args {
+    #[clap(long = "input", short = 'i', help = "Scenario file (default: stdin)")]
+    scenario_file: Option<PathBuf>,
+    #[clap(long = "output", short = 'o', help = "Graphviz file (default: stdout")]
+    output_file: Option<PathBuf>,
+}
+
+fn main() {
+    tracing_subscriber::fmt()
+        .with_writer(std::io::stderr)
+        .with_max_level(tracing::Level::DEBUG)
+        .init();
+
+    let args = Args::parse();
+
+    let scenario = match args.scenario_file {
+        Some(path) => read_to_string(path).expect("Failed to read scenario file"),
+        None => {
+            let mut input = String::new();
+            std::io::stdin()
+                .read_to_string(&mut input)
+                .expect("Failed to read from stdin");
+            input.trim().to_string()
+        }
+    };
+
+    let scenario: Scenario =
+        serde_yaml::from_str(&scenario).expect("Failed to parse YAML scenario file");
+
+    let executable = Executable::build(Messages::new(), &scenario, false)
+        .expect("Failed to build execution graph");
+
+    let result = executable.render_graph();
```

Comment:
The problem with `validate_types: bool` would not exist if we'd use a `Scenario` to draw a graph.

I also think, inability to draw a graph for a `Scenario` that does not "compile" into an `Executable`, does not really make for a good experience.